tests/torch/test_resume_from_checkpoint.py

```diff
@@ -0,0 +1,286 @@
+                # resume works fine for magnitude <-> const combo, because they have similar parameters
+                if s != v and ('magnitude' in s and 'const' in v or 'const' in s and 'magnitude' in v):
+                    resume_ok = True
+
+    return {
+        'save_algos': save_algos,
+        'load_algos': load_algos,
+        'is_resume_ok': resume_ok
+    }
+
+
+MODEL_WRAPPER = ["CPU", "GPU"]
+WRAPPERS = list(itertools.product(MODEL_WRAPPER, MODEL_WRAPPER))
+
+
+@pytest.fixture(scope='function', params=WRAPPERS,
+                ids=['_'.join(['from:' + w[0], 'to:' + w[1]]) for w in WRAPPERS])
+def _model_wrapper(request):
+    modes = request.param
+
+    def wrap_model(mode, model):
+        if mode == "GPU":
+            model = DataParallel(model, [0])
+        return model
+
+    return {
+        'save_model': partial(wrap_model, modes[0]),
+        'resume_model': partial(wrap_model, modes[1]),
+    }
+
+
+@pytest.mark.parametrize('is_resume', (True, False), ids=['resume', 'load_weights'])
+def test_load_state_interoperability(_algos, _model_wrapper, is_resume):
+    config_save = get_empty_config()
+    config_save['compression'] = [{'algorithm': algo} for algo in _algos['save_algos']]
+    register_bn_adaptation_init_args(config_save)
+    compressed_model_save, _ = create_compressed_model_and_algo_for_test(BasicConvTestModel(), config_save)
+    model_save = _model_wrapper['save_model'](compressed_model_save)
+    saved_model_state = model_save.state_dict()
+    ref_num_loaded = len(saved_model_state)
+
+    config_resume = get_empty_config()
+    config_resume['compression'] = [{'algorithm': algo} for algo in _algos['load_algos']]
+    register_bn_adaptation_init_args(config_resume)
+    compressed_model_resume, _ = create_compressed_model_and_algo_for_test(BasicConvTestModel(), config_resume)
+    model_resume = _model_wrapper['resume_model'](compressed_model_resume)
+
+    if not is_resume or (is_resume and _algos['is_resume_ok']):
+        act_num_loaded = load_state(model_resume, saved_model_state, is_resume)
+
+        if ('magnitude_sparsity' in _algos['load_algos'] or 'const_sparsity' in _algos['load_algos']) \
+            and 'rb_sparsity' in _algos['save_algos']:
+            # no need to load _mask and _uniform
+            ref_num_loaded -= 2
+        assert act_num_loaded == ref_num_loaded
+    else:
+        with pytest.raises(RuntimeError):
+            load_state(model_resume, saved_model_state, is_resume)
+
+
+RESUME_ALGOS = list(itertools.product([QUANTIZATION], SPARSITY_ALGOS))  # Q + 3S
+RESUME_ALGOS += [[algo] for algo in SPARSITY_ALGOS]  # 3S
+RESUME_ALGOS += [[QUANTIZATION]]  # Q
+RESUME_ALGOS += [['EMPTY']]  # No Compression
+RESUME_ALGOS = list(itertools.product(RESUME_ALGOS, RESUME_ALGOS))
+NUM_PARAMS_PER_ALGO = {
+    QUANTIZATION: 8,
+    'magnitude_sparsity': 1,
+    'const_sparsity': 1,
+    'rb_sparsity': 3,
+    'EMPTY': 0
+}
+
```

Comment:
Same

Reply:
using combinations of all algorithms would lead to overhead in tests. if we need custom combination of algorithms, the best way to configure - explicitly do a product of lists.